pkg/master/master.go

```diff
@@ -588,8 +638,8 @@ func (m *Master) getExtensionResources(c *Config) map[string]rest.Storage {
 	}
 
 	storage := map[string]rest.Storage{}
-	if isEnabled("horizontalpodautoscalers") {
-		autoscalerStorage, autoscalerStatusStorage := horizontalpodautoscaleretcd.NewREST(dbClient("horizontalpodautoscalers"), storageDecorator)
+	// Install only when appriopriate storages are provided.
+	if isEnabled("horizontalpodautoscalers") && autoscalerStorage != nil && autoscalerStatusStorage != nil {
 		storage["horizontalpodautoscalers"] = autoscalerStorage
 		storage["horizontalpodautoscalers/status"] = autoscalerStatusStorage
```

Comment:
Are you making the extensions/hpa resource return objects of type autoscaling/hpa? Because that will break old clients.